lib/index.js

```diff
@@ -469,6 +471,74 @@ class AuthenticationPlugin {
     return this.getUsersRepository().delete(user._id, {refresh: 'wait_for'});
   }
 
+  /**
+   * Search users by ES query concerning exclusively their username
+   *
+   * @param {Object} searchBody - Search request body (query, sort, etc.)
+   * @param {Object} options - from (undefined), size (undefined)
+   *
+   * @returns {Promise.<{ hits, total }>}
+   */
+  async search (searchBody, { from, size } = {}) {
+    debug('exists(searchBody = %j)', searchBody);
+
+    let sort;
+    let query;
+    let isRequiredFieldsInBody = false;
+    const forbiddenFields = extractESMappingFields(storageMapping.users);
+    const forbiddenKeywords = [
+      'multi_match',
+    ]; // @todo Keep up to date this blacklist to prevent queries on forbiddenFields
+
+    const sanitizeSearchBody = (key, value) => {
+      if (key === 'username') {
+        isRequiredFieldsInBody = true;
+        return { key: '_id' };
+      }
+      else if (value === 'username') {
+        isRequiredFieldsInBody = true;
+        return { value: '_id' };
+      }
+      else if (forbiddenKeywords.includes(key)) {
+        throw new this.context.errors.BadRequestError(`The "${key}" keyword is not allowed in this search query for security concerns.`);
```

Comment:
You should import the `BadRequestError` directly from kuzzle package instead
(apply to other errors)

Reply:
@MathieuVeber Here you should not throw the error from the PluginContext, this behavior is deprecated